packages/drivers/routerlicious-driver/src/documentServiceFactory.ts

```diff
@@ -76,49 +78,26 @@ export class RouterliciousDocumentServiceFactory implements IDocumentServiceFact
         const rateLimiter = new RateLimiter(this.driverPolicies.maxConcurrentOrdererRequests);
         const ordererRestWrapper = await RouterliciousOrdererRestWrapper.load(
             tenantId,
-            id,
+            undefined,
             this.tokenProvider,
             logger2,
             rateLimiter,
-            resolvedUrl.endpoints.ordererUrl,
+            createNewResolvedUrl.endpoints.ordererUrl,
         );
         const documentId = await ordererRestWrapper.post<string>(
             `/documents/${tenantId}`,
             {
-                id,
                 summary: convertSummaryToCreateNewSummary(appSummary),
                 sequenceNumber: documentAttributes.sequenceNumber,
                 values: quorumValues,
             },
         );
-        /**
-         * Assume documentId is at end of url path.
-         * This is true for Routerlicious' and Tinylicious' documentUrl and deltaStorageUrl.
-         * Routerlicious and Tinylicious do not use documentId in storageUrl nor ordererUrl.
-         * TODO: Ideally we would be able to regenerate the resolvedUrl, rather than patching the current one.
-         */
-        const replaceDocumentIdInPath = (urlPath: string): string =>
-            urlPath.split("/").slice(0, -1).concat([documentId]).join("/");
-        parsedUrl.pathname = replaceDocumentIdInPath(parsedUrl.pathname);
-        const deltaStorageUrl = resolvedUrl.endpoints.deltaStorageUrl;
-        if (!deltaStorageUrl) {
-            throw new Error(
-                `All endpoints urls must be provided. [deltaStorageUrl:${deltaStorageUrl}]`);
-        }
-        const parsedDeltaStorageUrl = parse(deltaStorageUrl);
-        parsedDeltaStorageUrl.pathname = replaceDocumentIdInPath(parsedUrl.pathname);
 
-        return this.createDocumentService(
-            {
-                ...resolvedUrl,
-                url: parsedUrl.href,
-                id: documentId,
-                endpoints: {
-                    ...resolvedUrl.endpoints,
-                    deltaStorageUrl: parsedDeltaStorageUrl.href,
-                },
-            },
-            logger);
+        // Craft a resolved URL using the new document ID as an URL
+        const resolvedUrl = await this.urlResolver.resolve({ url: documentId });
```

Comment:
this isn't a valid call to a the url resolver, as url resolvers should expect fully qualified urls. documentId isn't nearly enough to resolve arbitratily. to me this points to a ton of hidden assumptions about the expectations of the urls resolver used her. i'd much rather see these be explict.

Reply:
Yes, it's built on top of r11s specific assumptions. Currently the driver parses urls assuming certain format and order of parameters.

Reply:
these assumptions are not part of the url resolver interface. i think these assumptions should be made explicit